### DISCUSSION TITLE
0.64.0

### DISCUSSION THREAD

**Discussion by benhillis** [microsoft/WSL#8641] [General]
* Implement new error model
    * Localized error messages with inserts
    * Contextualized error messages
    * String readable HRESULTs
    
    Sample error:
    ```
    C:\Users\Test>wsl --mount A
    Administrator access is needed to mount a disk.
    Error code: Wsl/Service/AttachDisk/WSL_E_ELEVATION_NEEDED_TO_MOUNT_DISK
    ```

* Manually reset the interop server when the relay exits [[GH 7506](https://github.com/microsoft/wsl/issues)]
* Ensure WSLg logs will be present in /mnt/wslg/stderr.log
* WSL1 errors will now be printed to /dev/kmsg
* Add opt-in environment variable to use UTF8 encoding
    * Adds the opt-in `WSL_UTF8=1` environment variable to force UTF8 encoding on stdout & stderr through the CRT
    * Get rid of the double \r in our error messages

* Update WSLg to version 1.0.40  
    * libweston-desktop/xwayland: Use correct geometry
    * xwm: Generate more synthetic ConfigureNotify events
    * xwayland: Support _NET_FRAME_EXTENTS
    * rdp: fix window doesn't redraw properly moving across different DPI monitors
    * rdp: Move audio processing to front-end
    * rdp: Clipboard changes (sync with upstream)
    * WSLGd: don't override certain env if already set
    * Build: fix cut&paste mistake for DirectX-headers git version
    * rdp-plugin: various bug fixes in WSLDVCPlugin DLL

<hr /><em>This discussion was created from the release <a href='https://github.com/microsoft/WSL/releases/tag/0.64.0'>0.64.0</a>.</em>

**Comment by NotTheDr01ds** [comment]
@benhillis 

> Add opt-in environment variable to use UTF8 encoding

Thank you (and/or the team member who fixed this, but I get the feeling from some of your past changes that this is your area of code)!!!

Note that you can probably close #4607 as `fixinbound` based on this, right?  Or is this just considered an "opt-in-workaround" rather than a fix?

Funny, I've referenced that issue so many times I remembered the number without having to look it up ... ;-)

**Comment by orldiaz** [comment]
If I run the terminal as admin it says Invalid command line option: --mount so I cannot mount the disk on WSL

**Reply by NotTheDr01ds** [reply]
Wheat does `wsl --version` show?  Might be the same message, indicating an older release of WSL.  Does it show the same under both Admin and normal terminals?
